lab_2_tokenize_by_bpe/main.py

```diff
@@ -81,6 +189,23 @@ def decode(
     :param end_of_word_token: an end-of-word token
     :return: decoded sequence
     """
+    if not isinstance(encoded_text, list) or not isinstance(vocabulary, dict) or not (isinstance(
+            end_of_word_token, str) or end_of_word_token is None):
+        return None
+    decoded = ''
+    for identifier in encoded_text:
+        token_list = [key for key in vocabulary if vocabulary[key] == identifier]
+
+        for token in token_list:
+            if end_of_word_token:
+                if end_of_word_token in token:
```

Comment:
do `decoded += token` always. And add ' ' only sometimes.

Reply:
done